### DISCUSSION TITLE
Lesson 2: Multi-limb SplitNN on MNIST: 'Request to access data length rejected' error on final epoch training step

### DISCUSSION THREAD

**Discussion by cbs-rschreijen** [OpenMined/courses#305] [Course: Foundations of Private Computation]
Hi, 

I am getting the following error when running the final epoch training cell in the DS notebook. All previous cells in DO1, DO2 and DS notebooks ran error-free. I downloaded the most recent syft versions as per course instruction.

[2021-04-02T14:21:42.184665+0200][CRITICAL][logger]][29845] Request to access data length rejected.
[2021-04-02T14:21:42.185500+0200][CRITICAL][logger]][29845] Request to access data length rejected.
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/klass.py in __len__(self)
    255                 if data_len is None:
--> 256                     raise Exception
    257 

Exception: 

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/klass.py in __iter__(self)
    222             try:
--> 223                 data_len = self.__len__()
    224             except Exception:

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/klass.py in __len__(self)
    259             except Exception:
--> 260                 traceback_and_raise(
    261                     ValueError("Request to access data length rejected.")

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/logger.py in traceback_and_raise(e, verbose)
     60         e = Exception(e)
---> 61     raise e
     62 

ValueError: Request to access data length rejected.

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-11-230656f1092f> in <module>
      3     train_total = 0
      4 
----> 5     for in1, in2, label in zip(dl_1, dl_2, dl_local):
      6         opt1.zero_grad()
      7         opt2.zero_grad()

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/ast/klass.py in __iter__(self)
    223                 data_len = self.__len__()
    224             except Exception:
--> 225                 traceback_and_raise(
    226                     ValueError("Request to access data length rejected.")
    227                 )

~/opt/anaconda3/envs/conda-env/lib/python3.8/site-packages/syft/logger.py in traceback_and_raise(e, verbose)
     59     if not issubclass(type(e), Exception):
     60         e = Exception(e)
---> 61     raise e
     62 
     63 

ValueError: Request to access data length rejected.

**Comment by gonzalo-munillag** [comment]
I had something similar @cbs-rschreijen, but about the array lengths. I was not able to solve it. You can find the error here: https://github.com/OpenMined/courses/discussions/303

**Comment by znreza** [comment]
Hi @cbs-rschreijen if the issue is still not solved, could you please share it in the #support channel in openmined.slack.com? Someone will get back to you.

**Reply by josecyc** [reply]
I've shared it in the support channel, but no answer, I ran into the same issue.

**Reply by krisdobrev** [reply]
Has anyone been able to fix that issue?

**Comment by jvlaanen** [comment]
Hi all, I have encountered the same problem and the following solution worked for me: 

For your data owners, change
`duet.requests.add_handler(action="accept", tags=[""])`

to 

`duet.requests.add_handler(action="accept")`

Run this cell again for both data owners and you should be able to train your model. 

Good luck!
